### PR TITLE
Vendors now will say "Thanks for using!"

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

All vendor now will say `Thanks for using {vendor name}!` after dispensing an item
## Why / Balance

cool
## Media

https://github.com/space-wizards/space-station-14/assets/124214523/c9d4b68f-da5b-4550-b4f4-fa368da30305
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Vendors will thank you now for using them.

### CODE REVIEW

File: Content.Server/VendingMachines/VendingMachineSystem.cs [hunk 1]
```diff
@@ -384,6 +386,13 @@ private void EjectItem(EntityUid uid, VendingMachineComponent? vendComponent = n
                 _throwingSystem.TryThrow(ent, direction, vendComponent.NonLimitedEjectForce);
             }
 
+            if (!TryComp<MetaDataComponent>(uid, out var meta))
```
Comment (deltanedas) [inline Content.Server/VendingMachines/VendingMachineSystem.cs:389]: replace meme with Name(uid), metadata is always on every object

**Comment by PoorMansDreams** [comment]
does it thank you **everytime** you get an item?

**Comment by IamVelcroboy** [comment]
Neat. Could I suggest, it has an alternate message when e-magged?

**Comment by chromiumboy** [comment]
This seems like it'd get very spammy and annoying pretty quickly

**Comment by UbaserB** [comment]
I cant think of any way to tweak this without being abusable for spam or just generally not be annoying

**Comment by qwerltaz** [comment]
How about only sending this after client closes the UI if the vendor dispensed at least one item

File: Content.Server/VendingMachines/VendingMachineSystem.cs [hunk 2]
```diff
@@ -384,6 +386,9 @@ private void EjectItem(EntityUid uid, VendingMachineComponent? vendComponent = n
                 _throwingSystem.TryThrow(ent, direction, vendComponent.NonLimitedEjectForce);
             }
 
+            // Send message after dispensing
+                _chat.TrySendInGameICMessage(uid, Loc.GetString("vending-machine-thanks", ("name", Name(uid))), InGameICChatType.Speak, true);
+
             vendComponent.NextItemToEject = null;
             vendComponent.ThrowNextItem = false;
```
Comment (EmoGarbage404) [inline Content.Server/VendingMachines/VendingMachineSystem.cs:393]:
```suggestion
            _chat.TrySendInGameICMessage(uid, Loc.GetString("vending-machine-thanks", ("name", Name(uid))), InGameICChatType.Speak, true);

            vendComponent.NextItemToEject = null;
            vendComponent.ThrowNextItem = false;
```

**Comment by chromiumboy** [comment]
@lzk228, the vendor spam is a bit much. Would you mind updating this in line with qwerltaz's suggestion?

Also, the thank you message also plays for some vending machines that don't speak, like gas tank dispensers

**Comment by lzk228** [comment]
> Also, the thank you message also plays for some vending machines that don't speak, like gas tank dispensers

thats can be done with checking for advertise component



> Would you mind updating this in line with qwerltaz's suggestion?

dunno how to do this

**Comment by chromiumboy** [comment]
> dunno how to do this

No worries, I'll open an issue for it, if it doesn't exist yet

**Pull request merged**